tests/driver_at86rf2xx_aes/main.c

```diff
@@ -0,0 +1,230 @@
+}
+
+static void _cbc(at86rf2xx_t *dev,
+                 aes_block_t *plain,
+                 uint8_t key[AT86RF2XX_AES_BLOCK_SIZE],
+                 uint8_t iv[AT86RF2XX_AES_BLOCK_SIZE],
+                 aes_block_t *cipher,
+                 uint8_t nblocks)
+{
+    at86rf2xx_aes_key_write_encrypt(dev, key);
+    at86rf2xx_aes_cbc_encrypt(dev, cipher, key, iv, plain, nblocks);
+    memset(plain, 0, AT86RF2XX_AES_BLOCK_SIZE * nblocks);
+    at86rf2xx_aes_key_write_decrypt(dev, key);
+    at86rf2xx_aes_cbc_decrypt(dev, plain, key, iv, cipher, nblocks);
+}
+
+int main(void)
+{
+    xtimer_init();
+    at86rf2xx_t dev;
+    at86rf2xx_setup(&dev, &at86rf2xx_params[0]);
+    dev.netdev.netdev.event_callback = _event_cb;
+    if (dev.netdev.netdev.driver->init(&dev.netdev.netdev) != 0) {
+        return EXIT_FAILURE;
+    }
+
+    {
+        uint8_t k[] = KEY_1;
+        uint8_t p[] = PLAIN_1;
+        uint8_t c[sizeof(p)];
+        uint8_t n = sizeof(p) / AT86RF2XX_AES_BLOCK_SIZE;
+
+        _ecb(&dev, (aes_block_t *)p, k, (aes_block_t *)c, n);
+
+        if (!memcmp(p, (uint8_t[])PLAIN_1, sizeof(p))) {
+            puts("ECB test 1 done");
+        }
+        else {
+            puts("ECB test 1 failed");
+        }
+    }
+    {
+        uint8_t k[] = KEY_2;
+        uint8_t p[] = PLAIN_2;
+        uint8_t c[sizeof(p)];
+        uint8_t n = sizeof(p) / AT86RF2XX_AES_BLOCK_SIZE;
+
+         _ecb(&dev, (aes_block_t *)p, k, (aes_block_t *)c, n);
+
+        if (!memcmp(p, (uint8_t[])PLAIN_2, sizeof(p))) {
+            puts("ECB test 2 done");
+        }
+        else {
+            puts("ECB test 2 failed");
+        }
+    }
+    {
+        uint8_t k[] = KEY_3;
+        uint8_t v[] = IV_3;
+        uint8_t p[] = PLAIN_3;
+        uint8_t c[sizeof(p)];
+        uint8_t n = sizeof(p) / AT86RF2XX_AES_BLOCK_SIZE;
+
+        _cbc(&dev, (aes_block_t *)p, k, v, (aes_block_t *)c, n);
+
+        if (!memcmp(p, (uint8_t[])PLAIN_3, sizeof(p))) {
+            puts("CBC test 1 done");
+        }
+        else {
+            puts("CBC test 1 failed");
+        }
+    }
+    {
+        uint8_t k[] = KEY_4;
+        uint8_t v[] = IV_4;
+        uint8_t p[] = PLAIN_4;
+        uint8_t c[sizeof(p)];
+        uint8_t n = sizeof(p) / AT86RF2XX_AES_BLOCK_SIZE;
+
+        _cbc(&dev, (aes_block_t *)p, k, v, (aes_block_t *)c, n);
+
+        if (!memcmp(p, (uint8_t[])PLAIN_4, sizeof(p))) {
+            puts("CBC test 2 done");
+        }
+        else {
+            puts("CBC test 2 failed");
+        }
+    }
+    {
+        uint8_t k[] = KEY_5;
+        uint8_t v[] = IV_5;
+        uint8_t p[] = PLAIN_5;
+        uint8_t c[sizeof(p)];
+        uint8_t n = sizeof(p) / AT86RF2XX_AES_BLOCK_SIZE;
+
+        _cbc(&dev, (aes_block_t *)p, k, v, (aes_block_t *)c, n);
+
+        if (!memcmp(p, (uint8_t[])PLAIN_5, sizeof(p))) {
+            puts("CBC test 3 done");
+        }
+        else {
+            puts("CBC test 3 failed");
+        }
+    }
```

Comment:
Would be good if the test would print `SUCCESS` at the end.
Then you can also add a `tests/01-run.py` with

```Python
#!/usr/bin/env python3

import sys
from testrunner import run

def testfunc(child):
    child.expect('START')
    child.expect('ECB test 1 done')
    child.expect('ECB test 2 done')
    child.expect('CBC test 1 done')
    child.expect('CBC test 2 done')
    child.expect('CBC test 3 done')
    child.expect('SUCCESS')

if __name__ == "__main__":
    sys.exit(run(testfunc))
```

so we can have `make test` work for automatic tests.

Reply:
Ok

Reply:
Hm looks like you'll have to add a `puts("START");` at the beginning of the test too.
This and `chmod a+x tests/01-run.py`.

Then `make test` will work.

Reply:
Or just remove matching `start`, `test_utilis_interactive_sync` will already match the start for you.

Reply:
ups forgot that ...